lib/k8s/resource_client.rb

```diff
@@ -162,20 +169,22 @@ def get?
     # @param namespace [String, NilClass]
     # @return [Object] instance of resource_class
     def get(name, namespace: @namespace)
+      response_class = response_class(path('{name}', namespace: '{namespace}')) || @resource_class
       @transport.request(
         method: 'GET',
         path: path(name, namespace: namespace),
-        response_class: @resource_class
+        response_class: response_class
       )
     end
 
     # @param resource [resource_class]
     # @return [Object] instance of resource_class
     def get_resource(resource)
+      response_class = response_class('{name}', namespace: '{namespace}') || @resource_class
```

Comment:
this one does `response_class('{name}')` and the previous one does `response_class(path('{name}'))`, on purpose?

Reply:
It's a bug.